core/src/main/java/org/elasticsearch/bootstrap/Bootstrap.java

```diff
@@ -190,7 +180,7 @@ public void run() {
             Runtime.getRuntime().addShutdownHook(new Thread() {
                 @Override
                 public void run() {
-                    try {
+                    try (Spawner spawnerToClose = spawner) {
```

Comment:
We should keep logging as the last component to shutdown, maybe we should do this:

```diff
diff --git a/core/src/main/java/org/elasticsearch/bootstrap/Bootstrap.java b/core/src/main/java/org/elasticsearch/bootstrap/Bootstrap.java
index 3bfc2ac3b5..1e53faa9ef 100644
--- a/core/src/main/java/org/elasticsearch/bootstrap/Bootstrap.java
+++ b/core/src/main/java/org/elasticsearch/bootstrap/Bootstrap.java
@@ -180,8 +180,8 @@ final class Bootstrap {
             Runtime.getRuntime().addShutdownHook(new Thread() {
                 @Override
                 public void run() {
-                    try (Spawner spawnerToClose = spawner) {
-                        IOUtils.close(node);
+                    try {
+                        IOUtils.close(node, spawner);
                         LoggerContext context = (LoggerContext) LogManager.getContext(false);
                         Configurator.shutdown(context);
                     } catch (IOException ex) {
@@ -258,8 +258,8 @@ final class Bootstrap {
     }
 
     static void stop() throws IOException {
-        try (Spawner spawnerToClose = INSTANCE.spawner) {
-            IOUtils.close(INSTANCE.node);
+        try {
+            IOUtils.close(INSTANCE.node, INSTANCE.spawner);
         } finally {
             INSTANCE.keepAliveLatch.countDown();
         }
```

Also, we are probably missing a logging shutdown in the `stop` method, but let's worry about that separately.